### DISCUSSION TITLE
Buttons that look like Anchors

### DISCUSSION THREAD

**Discussion by eshanholtz** [twilio-labs/paste#2428] [Q&A] [answered]
Hi Team! 

My question seems like it may be related to an issue discussed [here](https://github.com/twilio-labs/paste/discussions/1820#discussion-3548459).

I've run into a situation where I need a button to look like an Anchor. For the migration of the Error Logs page to Console.zen, we have a scenario where in the "Individual Debug Events" view (first picture below), the event name is an Anchor that redirects to a new URL. However, in the "Aggregated Events" view (second picture below), we want the Event's text to _look_ like an Anchor, so that the UI is consistent, but clicking on the event does not redirect to a new URL -- it performs an in-page action. Based on the criteria Paste has laid out regarding the difference between Anchors and Buttons, a Button is more appropriate for the text in the second view, but the "link" variant on the Button leads to the text being bolded and there is no underline on the text unless hovered over. 

If there is no plan to add an "anchor" variant to the Button API, would it be appropriate here to use an Anchor component with an "onClick" prop and that does not redirect to a new URL, or is that "breaking the rules"?

Individual Debug Event view
![Screen Shot 2022-05-23 at 3 14 57 PM](https://user-images.githubusercontent.com/28262845/169913870-37fa60fc-582e-4624-8472-f24dc798b95d.png)

Aggregated Events view
![Screen Shot 2022-05-23 at 3 14 42 PM](https://user-images.githubusercontent.com/28262845/169913848-0f4700d7-6f2d-44a4-9d46-8d7f55608107.png)

**Comment by nkrantz** [comment]
Hi @eshanholtz - the designs in the two screenshots you shared seem to have similar appearances. Are you showing what it looks like to use Anchors in both views? To help provide more context, could you share a screenshot of what the visual differences look like that you're trying to avoid (when you're using the appropriate components)?

The reason it's a bad practice to use an Anchor (when clicking on the element doesn't redirect to a new URL) is that you're incorrectly communicating to some of your users what their action is going to do. When there's a clickable `<a>` component, the information communicated to screen reader users tells them that performing that action will lead to a new page. It's confusing and makes pages difficult to use when there are inconsistencies across the components and their intended usages.

In addition to the screenshots, could you provide some context about why it's important for the two pages to have the same style between the buttons and anchors? Is this based on feedback showing users are confused or dissatisfied with the visual differences?

Thanks!

**Reply by eshanholtz** [reply]
The screenshots I provided are the Figmas I'm working off of. Let me upload some screenshots/videos of the implemented UX when I use Anchor vs. Button (with "link" variant)

**Reply by eshanholtz** [reply]
https://user-images.githubusercontent.com/28262845/170148650-d1f7b23f-3a82-49ca-94f0-fe2d3355b4d4.mov

https://user-images.githubusercontent.com/28262845/170148654-35c45d13-8347-496d-83ba-468780fab24d.mov

**Reply by nkrantz** [reply]
Thanks for that context @eshanholtz. Just to be clear, are the Anchors and Buttons ever on the same screen together? Like, if you are using Link-style Buttons for the in-page actions, and you select "View all" and scroll down, are both link styles on the table at the same time? It would be helpful to see that view with them both on the page at the same time.

Also, what exactly happens when the user clicks an error code? What is the in-page action that happens?

Feel free to share the Figma link you're working from if that's easier than sharing screenshots/screen recordings. Thanks!

**Reply by eshanholtz** [reply]
@nkrantz  The Event names will not be a combination Anchors and Buttons within the Table at the same time. These items link to their "instance" page when "View all" is selected, so an Anchor is always appropriate here. When any other view is selected, clicking the item performs the in-page action of updating the query value + resetting the view to "View all" (shown below). However, the table in the "View all" view contains an "Inspect" Button with the "link" style that performs an in-page action of opening a Modal. The "Inspect" Button is not available in any other view. You can see a mock up of what this button will look like in the Table in the first screen shot of the Figma I posted with the original question, but it has not been implemented yet.

https://user-images.githubusercontent.com/28262845/170377062-a8326b60-04bf-4b14-b271-0637abe8f5e1.mov

https://user-images.githubusercontent.com/28262845/170377092-965fb76b-3df8-476e-9ec6-e7632c0a7238.mov


Figma link: https://www.figma.com/file/3GaINOa9Epx7ENTZslfaWa/Alarms-H1-2021?node-id=1840%3A3

**Comment by gloriliale** [comment]
Hi [eshanholtz](https://github.com/eshanholtz)! Thanks for including the video showing the in-page action that follows the button being clicked; it's super helpful knowledge. I've discussed it with the team, and we have a few ideas and recommendations that we think would make the interaction more user-friendly. Would it be possible to have you and the designer(s) you've been working with to come to one of our [upcoming office hours](https://docs.google.com/spreadsheets/d/10vD91zdwr9hjgDOV3HSGnoC0LgvYw-thmn3X_kvW0xk/edit#gid=664218049)?

**Reply by btjfsu** [reply]
@eshanholtz is on PTO this week, but I have signed up for office hours today (May 31st) on her behalf to discuss this further. Thanks!

**Comment by loreina** [comment, marked as answer]
Hi @btjfsu, thanks for coming to office hours! You can find a link to the recording in #help-design-system. If you have other questions, feel free to reply in this discussion or sign up for another office hours. Thanks!
